### ISSUE TITLE
How to get the bot to craft ingame

### ISSUE THREAD

**Issue by swantthetank** [kolton/d2bot-with-kolbot#1080]
Think Ive enabled it all and this is my error 
"Error in Crafting (crafting.js #16) Bad Crafting System config. (Area: 109, Ping:78, Game: Gamename)"

**Comment by semirotta** [comment]
That means there is something weird you're trying to craft. Did you add your crafts yourself or use the ones in config already? if you used the ones in config already, your crafts should work.
Can you hastebin.com your character config.

**Comment by swantthetank** [comment]
https://pastebin.com/22XYgn1d here it is

**Comment by swantthetank** [comment]
> That means there is something weird you're trying to craft. Did you add your crafts yourself or use the ones in config already? if you used the ones in config already, your crafts should work.
> Can you hastebin.com your character config.

Nah I didnt add anything myself I have checked to it is the right numbers, I mean the right ids for jewel amu ral and pame. It is cubig flawless to pame though. But its not making any amulets.

**Comment by semirotta** [comment]
>         Config.Recipes.push([Recipe.Rune, "Io Rune"]); // Upgrade Io to Lum
        Config.Recipes.push([Recipe.Rune, "Lum Rune"]); // Upgrade Lum to Ko
        Config.Recipes.push([Recipe.Rune, "Ko Rune"]); // Upgrade Ko to Fal
        Config.Recipes.push([Recipe.Rune, "Fal Rune"]); // Upgrade Fal to Lem
        Config.Recipes.push([Recipe.Rune, "Lem Rune"]); // Upgrade Lem to Pul
        Config.Recipes.push([Recipe.Rune, "Pul Rune"]); // Upgrade Pul to Um

Those seem to have weird ass space in front so you might want to re-check your config. Most likely something like this is causing the issue.

**Comment by swantthetank** [comment]
> > ```
> >     Config.Recipes.push([Recipe.Rune, "Io Rune"]); // Upgrade Io to Lum
> > ```
> 
> ```
>     Config.Recipes.push([Recipe.Rune, "Lum Rune"]); // Upgrade Lum to Ko
>     Config.Recipes.push([Recipe.Rune, "Ko Rune"]); // Upgrade Ko to Fal
>     Config.Recipes.push([Recipe.Rune, "Fal Rune"]); // Upgrade Fal to Lem
>     Config.Recipes.push([Recipe.Rune, "Lem Rune"]); // Upgrade Lem to Pul
>     Config.Recipes.push([Recipe.Rune, "Pul Rune"]); // Upgrade Pul to Um
> ```
> Those seem to have weird ass space in front so you might want to re-check your config. Most likely something like this is causing the issue.

Oh yeah you're right, but I added those after anyway. I had the problem before the errors on crafting started.

**Comment by semirotta** [comment]
Hm well that is indeed weird, I tried to go through the recipes again but I couldn't find anything other weird than those spaces. Do you know what it tries to craft when you get the error or is it happening no matter what you craft?
I have used the crafting for like a week now and I haven't had any issues. That's why I would imagine it being an issue in config.
Unfortunately I can't help more than this :( I'm a bit of a noob with this.

**Comment by swantthetank** [comment]
hmm he did actually craft one amu the first run now but the 2nd run I got the same error. Is it a way to keep all the crafted amus so i can decide if I want to save them or not?

**Comment by semirotta** [comment]
The bot ditches the bad amulets. You could try adding your personal pickit lines but thats most likely alot of work to add different kinds of amulets there since you can never know what you get I suppose.

**Comment by swantthetank** [comment]
yeah I know, but is it a way to see in the config what amulets he crafts? So i know that he only akara the bad ones :P

**Comment by JohnGoblikon** [comment]
> hmm he did actually craft one amu the first run now but the 2nd run I got the same error. Is it a way to keep all the crafted amus so i can decide if I want to save them or not?

I just have something like
[Type] == amulet && [Quality] == crafted # [ItemAddClassSkills] >= 2 
that way it saves all +2 skill crafted ammy sort through them later
or you could do
[Type] == amulet && [Quality] == crafted # [FCR] >= 5
Since all crafted caster ammys have at least 5% fcr all ammys crafted should be saved

**Comment by swantthetank** [comment]
> > hmm he did actually craft one amu the first run now but the 2nd run I got the same error. Is it a way to keep all the crafted amus so i can decide if I want to save them or not?
> 
> I just have something like
> [Type] == amulet && [Quality] == crafted # [ItemAddClassSkills] >= 2
> that way it saves all +2 skill crafted ammy sort through them later
> or you could do
> [Type] == amulet && [Quality] == crafted # [FCR] >= 5
> Since all crafted caster ammys have at least 5% fcr all ammys crafted should be saved

oh yeah you're right that's kinda nice. Why didnt I think about that hehe

**Comment by semirotta** [comment]
So basically you could just have the skills 2 and fcr like 10 so it would keep the 2 skilled with 10 fcr. On top of that you could add resistances, str / dex etc.

**Comment by JohnGoblikon** [comment]
[Type] == amulet && [Quality] == crafted # [ItemAddClassSkills] >= 2 && [FCR] == 10
this would only keep the +2 skill and 10fcr not >10 or <10
I just left mine at >=5 because you might not need 10+ fcr to reach breakpoints for all builds

**Comment by aim2kill** [comment]
https://pastebin.com/eVc2aSWk <<so so crafted
https://pastebin.com/YP8HxGk2 <<better crafted

**Comment by swantthetank** [comment]
> edit to your liking
> https://pastebin.com/YP8HxGk2

There it saves all amus with +2 skill right? :p 
It seems like it does atleast :D

**Comment by aim2kill** [comment]
Yes line 40
[Type] == Amulet && [Quality] == Crafted # [ItemAddClassSkills] == 2

**Comment by watilove** [comment]
im really bad at coding can someone show me what should the  Craftingsystem.js page should look like to set for crafting caster amulets?  and what else should i know about it   thx guys
